src/main/java/seedu/address/model/patient/ApptList.java

```diff
@@ -55,6 +56,30 @@ public List<Appt> getAppts() {
         return Collections.unmodifiableList(appts);
     }
 
+    /**
+     * Returns the most recent past appointment.
+     * @return Appt
+     */
+    public Appt getMostRecentPastAppt() {
+        LocalDateTime now = LocalDateTime.now();
+        return appts.stream()
+                .filter(appt -> appt.getDateTime().isBefore(now))
+                .reduce((first, second) -> second)
+                .orElse(null);
+    }
+
+    /**
+     * Returns most recent future appointment.
+     * @return Appt
+     */
+    public Appt getMostRecentFutureAppt() {
+        LocalDateTime now = LocalDateTime.now();
+        return appts.stream()
+                .filter(appt -> appt.getDateTime().isAfter(now))
+                .findFirst()
+                .orElse(null);
+    }
+
```

Comment:
I think abstraction will be improved for Appointments with this change, thank you!